eyre/build.rs

```diff
@@ -26,59 +56,44 @@ const BACKTRACE_PROBE: &str = r#"
     }
 
     impl Error for E {
-        fn backtrace(&self) -> Option<&Backtrace> {
-            let backtrace = Backtrace::capture();
-            match backtrace.status() {
-                BacktraceStatus::Captured | BacktraceStatus::Disabled | _ => {}
-            }
-            unimplemented!()
+        fn provide<'a>(&'a self, request: &mut Request<'a>) {
+            request
+                .provide_ref::<MyBacktrace>(&self.backtrace);
         }
     }
 "#;
 
-const TRACK_CALLER_PROBE: &str = r#"
-    #![allow(dead_code)]
-
-    #[track_caller]
-    fn foo() {
-        let _location = std::panic::Location::caller();
-    }
-"#;
+fn compile_probe(probe: &str) -> Option<ExitStatus> {
+    let rustc = env::var_os("RUSTC")?;
+    let out_dir = env::var_os("OUT_DIR")?;
+    let probefile = Path::new(&out_dir).join("probe.rs");
+    fs::write(&probefile, probe).ok()?;
 
-fn main() {
-    match compile_probe(BACKTRACE_PROBE) {
-        Some(status) if status.success() => println!("cargo:rustc-cfg=backtrace"),
-        _ => {}
-    }
+    // Supports invoking rustc thrugh a wrapper
+    let mut cmd = if let Some(wrapper) = env::var_os("RUSTC_WRAPPER") {
```

Comment:
It turns out there is also RUSTC_WORKSPACE_WRAPPER which should probably be honored as well.
[This](https://github.com/dtolnay/proc-macro2/blob/master/build.rs) is a reasonable template to start with.